energy_communities_cooperator/controllers/website_share_subscription_controller.py

```diff
@@ -0,0 +1,233 @@
+        return values
+
+    def _validate_request(self, values):
+        if not values["company"]:
+            values.update(
+                {"global_error": True, "error_msgs": [_("Company not found")]}
+            )
+        if not values["mode"]:
+            values.update({"global_error": True, "error_msgs": [_("Mode not found")]})
+        if values["product_ext_id"] and not values["product"]:
+            self.values.update(
+                {"global_error": True, "error_msgs": [_("Product not found")]}
+            )
+
+    def _render_page(self, values):
+        return request.render(
+            "energy_communities_cooperator.template_subscription_page",
+            values,
+        )
+
+    # getters
+    def _get_model_from_ext_id(self, model_name, field_name, ext_id):
+        if ext_id:
+            return request.env[model_name].sudo().search([(field_name, "=", ext_id)])
+        return False
+
+    def _get_products_from_category_and_company(self, product_categ_id, company_id):
+        return (
+            request.env["product.template"]
+            .sudo()
+            .search(
+                [
+                    ("categ_id", "=", product_categ_id),
+                    ("default_share_product", "=", True),
+                    ("company_id", "=", company_id),
+                ]
+            )
+        )
+
+    def _get_page_title(self, values):
+        return MAPPING__SUBSCRIPTION_MODE__DEFAULT_PAGE_TITLE[values["mode"]].format(
+            company_name=values["company"].comercial_name
+        )
+
+    def _get_page_headline(self, values):
+        return MAPPING__SUBSCRIPTION_MODE__DEFAULT_PAGE_HEADLINE[values["mode"]]
+
+    def _get_page_headline_fixed_transfer(self, values):
+        return MAPPING__SUBSCRIPTION_MODE__DEFAULT_PAGE_HEADLINE_FIXED_TRANSFER[
+            values["mode"]
+        ].format(
+            product_price=str(values["product"].list_price).replace(".", ","),
+            currency_symbol=values["currency_symbol"],
+        )
+
+    def _get_page_headline_fixed_sepa(self, values):
+        return MAPPING__SUBSCRIPTION_MODE__DEFAULT_PAGE_HEADLINE_FIXED_SEPA[
+            values["mode"]
+        ].format(
+            product_price=str(values["product"].list_price).replace(".", ","),
+            currency_symbol=values["currency_symbol"],
+        )
+
+    def _get_form_fields_values(self, data, path):
+        keys = path.split(".")
+        current = data
+        for key in keys:
+            if not isinstance(current, dict):
+                try:
+                    current = current.read()[0]  # Convert the record to a dictionary
+                except:
+                    return ""  # Return an empty string if the record cannot be converted to a dictionary
+            if key not in current:
+                return ""  # Return an empty string if the field is not found
+            current = current[key]
+        return current
+
+    def _get_form_fields(self, values):
```

Comment:
I should rename the following methods (`_get_form_fields`, `_get_custom_options`, `_get_custom_description`, ` _get_page_values`)  to `_update`, instead of `_get`, because they aren't returning anything, but they are updating `values` dictionary